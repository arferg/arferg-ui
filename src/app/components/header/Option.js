import React from "react"
import {Link} from "react-router-dom"
import { useToasts } from 'react-toast-notifications'


const Option = props => {
    const {
        details,
        onClick
    } = props

    const { addToast } = useToasts();

    return (
        <div className={"option"}
             onClick={() => onClick ? onClick() : addToast('Clicked Successfully', { appearance: 'success' })}>
            {details.label}
        </div>
        // <Link to={details.link} >
        //     <div className={"option"}
        //          onClick={() => addToast('Clicked Successfully', { appearance: 'success' })}>
        //         {details.label}
        //     </div>
        // </Link>
    )
}

export default Option