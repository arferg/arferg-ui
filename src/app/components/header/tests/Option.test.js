import React from "react";
// import "@testing-library/jest-dom"
import {ToastProvider, useToasts} from 'react-toast-notifications'
import {render, fireEvent} from "@testing-library/react"
import Option from "../Option"
import {BrowserRouter} from "react-router-dom";

jest.mock('react-toast-notifications', () => (
    {
        ...(jest.requireActual('react-toast-notifications')),
        useToasts: jest.fn()
    }
))
const addToastMock = jest.fn()

const getComponent = (props) =>
    <BrowserRouter>
        <ToastProvider>
            <Option
                details={{
                    link: "/cats",
                    label: "Cat Pics"
                }}
                {...props} />
        </ToastProvider>
    </BrowserRouter>

describe("Option Tests", () => {
    beforeEach(() => {
        jest.clearAllMocks();

        // after jest.fn() too
        useToasts.mockImplementation(() => ({
            addToast: addToastMock
        }))
    });

    describe("Should render ", () => {
        it("correctly with the label", () => {
            const {asFragment} = render(getComponent())
            expect(asFragment()).toMatchSnapshot()
        });
    })

    describe("Should call ", () => {
        it("addToast whenever the option is clicked", () => {
            const {getByText} = render(getComponent())
            const option = getByText("Cat Pics")

            fireEvent.click(option)
            expect(addToastMock).toHaveBeenCalledWith('Clicked Successfully', { appearance: 'success' })
        });

        it("onClick prop whenever provided and the option is clicked", () => {
            // spyOn only overrides for this test
            const alertMock = jest.spyOn(window,'alert').mockImplementation();

            const onClick = jest.fn()
            // const onClick = jest.fn(() => alert("hayooooo"))
            onClick.mockImplementation(() => alert("hayooooo"))
            // onClick.mockReturnValue("returned string")

            const {getByText} = render(getComponent({onClick}))
            const option = getByText("Cat Pics")

            fireEvent.click(option)
            expect(alertMock).toHaveBeenCalledWith("hayooooo")
        });
    })
})
