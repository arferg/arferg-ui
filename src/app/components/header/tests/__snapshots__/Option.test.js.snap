// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Tests Should render  correctly with the label 1`] = `
<DocumentFragment>
  <div
    class="option"
  >
    Cat Pics
  </div>
</DocumentFragment>
`;
